w01/test/gad/logging/TestLogging.java

```diff
@@ -0,0 +1,73 @@
+package gad.logging;
+
+import gad.ExampleResultImplementation;
+import gad.maze.Maze;
+import gad.maze.Walker;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestLogging {
+
+    @Test
+    @DisplayName("Logging - Artemis Example")
+    public void checkLoggingArtemisExample() {
+        boolean[][] maze = Maze.generateStandardMaze(10, 10);
+        int[] solutionX = new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 3, 4, 4, 4, 3, 2, 1, 2, 3, 4, 4, 5, 6, 6, 7, 8, 9 };
+        int[] solutionY = new int[] { 0, 1, 2, 3, 4, 5, 6, 5, 5, 5, 6, 6, 7, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 8, 8, 8, 8 };
+        ExampleResultImplementation result = new ExampleResultImplementation(solutionX.length);
+        Walker walker = new Walker(maze, result);
+        assertTrue(walker.walk(), "Walker should find exit");
+        result.testLogging(solutionX, solutionY);
+    }
```

Comment:
```suggestion
    @Test
    @DisplayName("Logging - Artemis Example")
    @MethodSource
    public void checkLogging(int width, int height, int seed, int[] solutionX, int[] solutionY, boolean expect) {
        boolean[][] maze = Maze.generateMaze(width, height, seed);
        
        ExampleResultImplementation result = new ExampleResultImplementation(solutionX.length);
        Walker walker = new Walker(maze, result);
        assertEquals(expect, walker.walk(), "Walker should" + (expect ? "" : " not") + " find exit");
        result.testLogging(solutionX, solutionY);
    }
    
    private static Stream<Arguments> checkLogging() {
        return Stream.of(
          Arguments.of(
              10, 10, 0
              new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 3, 4, 4, 4, 3, 2, 1, 2, 3, 4, 4, 5, 6, 6, 7, 8, 9 },
              new int[] { 0, 1, 2, 3, 4, 5, 6, 5, 5, 5, 6, 6, 7, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 8, 8, 8, 8 },
              true
          )
          //, More Arguments here
        );
    }
```

Reply:
I have changed my Tests to a parameterized structure.